src/main/kotlin/org/rust/lang/refactoring/extractFunction/RsExtractFunctionConfig.kt

```diff
@@ -76,26 +97,42 @@ class RsExtractFunctionConfig private constructor(
             val fn = first.ancestorStrict<RsFunction>() ?: return null
             if (fn != last.ancestorStrict<RsFunction>()) return null
 
-            val letBindings = fn.descendantsOfType<RsPatBinding>().filter { it.textOffset <= end }
+            val letBindings = fn.descendantsOfType<RsPatBinding>()
+                .filter { it.textOffset <= end }
 
-            val innerBinding = letBindings.filter { it.textOffset >= start }
+            val parameters = letBindings
+                .filter { it.textOffset <= start }
+                .filter {
+                    ReferencesSearch.search(it, LocalSearchScope(fn))
+                        .any { ref -> ref.element.textOffset in start..end }
```

Comment:
This filter doesn't take into account function arguments
```rust
fn foo(a: i32) {
    <selection>println!("{}", a);</selection>
}
```
=>
```rust
fn foo(a: i32) {
    bar();
}

fn bar() {
    println!("{}", a); // error[E0425]: cannot find value `a` in this scope
}
```

Reply:
Ups, i miss this.